### ISSUE TITLE
1K-step training run with checkpointing

### ISSUE THREAD

**Issue by Hmbown** [Hmbown/Butterfly#17]
## Goal
Real-ish 1K-step training run. Produce a checkpoint that loads back and generates (garbage is fine — just need a non-crashing round-trip).

**Depends on:** #16

## Acceptance criteria
- [ ] `checkpoints/butterfly_150m_step1000.pt` exists and is < 2 GB.
- [ ] Loss curve shows clear downward trend (not plateau at init).
- [ ] Small eval script loads the checkpoint + generates 50 tokens with non-NaN output.
- [ ] Log at `results/training/butterfly_150m_step1000.log`.

## Run
```bash
python scripts/train_150m_butterfly_cuda.py \
    --data data/nemotron_cc_train.bin \
    --max-steps 1000 \
    --batch-size 4 --grad-accum 4 \
    --log-every 20 \
    --ckpt-out checkpoints/butterfly_150m_step1000.pt \
    --ckpt-interval 500 \
    2>&1 | tee results/training/butterfly_150m_step1000.log
```

Then write `scripts/sample_butterfly_checkpoint.py` (~30 lines) that:
1. Loads `checkpoints/butterfly_150m_step1000.pt`.
2. Rebuilds the config from `ckpt["config"]`.
3. Loads the tokenizer from `ckpt["meta"]["tokenizer_type"]` (fallback gpt2).
4. Greedy-generates 50 tokens from a hardcoded prompt.
5. Prints the decoded string.

## Gotchas
- Confirm the ckpt has config + state_dict: `python -c "import torch; ckpt=torch.load('checkpoints/butterfly_150m_step1000.pt'); print(ckpt.keys())"`.
- 1000 steps × 32k tokens = 32M tokens seen — nowhere near full convergence; expect loss ~4–5, not < 2.

## Deliverable
Loss plot + 200-char generation sample + the sampling script.

**Estimate:** 2–6 hrs depending on throughput.
